openml-api/src/main/java/com/feedzai/openml/explanations/ModelExplainer.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022 Feedzai
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.feedzai.openml.explanations;
+
+import com.feedzai.openml.data.Instance;
+import com.feedzai.openml.model.MachineLearningModel;
+
+/**
+ * A base class for explanation algorithms. Holds any {@link MachineLearningModel} and can compute the
+ * feature contributes for its predictions through {@link #getFeatureContributions(Instance)}.
+ *
+ * @param <M> Each {@link ModelExplainer} must be associated with an implementation of {@link MachineLearningModel}.
+ * @since 1.1.1
+ */
+public interface ModelExplainer<M extends MachineLearningModel> {
```

Comment:
ModelExplainer<M extends MachineLearningModel>

seems weird when the generic type is not being used.

If we want each implementation to be coupled with a specific model, I think Paulo's approach is better.

Buf it you truly want it to be model agnostic, why not have just `public interface ModelExplainer` ? 
I get that it will be used on the implementations like `FzRandomForestExplainer implements ModelExplainer<FzRandomForest> `or something like that, but is there any real advantages in doing this?

Reply:
You are absolutely right. It was because we used to have a getModel method and we didn't rethink it.